components/ParameterForm.vue

```diff
@@ -118,27 +126,36 @@ const renderAsRadios = (parameter: Parameter) => {
   return parameter.parameterType === ParameterType.Select && optionsAreTerse(parameter);
 };
 
-const submitForm = () => {
-  // Not implemented yet
-};
+const formSubmitting = ref(false);
 
-const largeScreen = ref(true);
-const breakpoint = 992; // CoreUI's "lg" breakpoint
-const setFieldSizes = () => {
-  if (window.innerWidth < breakpoint) {
-    largeScreen.value = false;
-  } else {
-    largeScreen.value = true;
-  }
+const submitForm = async () => {
+  if (!formData.value) {
+    return;
+  };
+
+  formSubmitting.value = true;
+  const response = await $fetch<NewScenarioData>("/api/scenarios", {
+    method: "POST",
+    query: { // Using query instead of body because I couldn't work out how to send a body in the integration test.
+      parameters: formData.value,
+    },
+  }).catch((error: FetchError) => {
+    console.error(error);
+  });
+
+  if (response) {
+    const { runId } = response;
+    navigateToData.value = `/scenarios/${runId}`;
+    await navigateTo(navigateToData.value);
+  };
 };
 
 onMounted(() => {
-  setFieldSizes();
-  window.addEventListener("resize", setFieldSizes);
-});
-onBeforeUnmount(() => {
-  window.removeEventListener("resize", setFieldSizes);
+  pageMounted.value = true;
 });
+
+const appStore = useAppStore();
```

Comment:
should move this at top with all variable definitions... nice to keep nicely separated with methods